crates/trippy-tui/src/locale.rs

```diff
@@ -1,80 +1,38 @@
-const FALLBACK_LOCALE: &str = "en";
+use i18n_embed::{fluent::FluentLanguageLoader, DesktopLanguageRequester, LanguageRequester};
+use rust_embed::RustEmbed;
+use std::sync::OnceLock;
 
-/// Set the locale for the application.
-///
-/// If the given locale is `None` or unsupported, the system locale is tried. If the system locale
-/// is not supported, the fallback locale is used.
-///
-/// In both cases, the language part of the locale is used if the full locale is not supported.
-pub fn set_locale(locale: Option<&str>) {
-    if let Some(locale) = locale {
-        set_locale_inner(locale);
-    } else if let Some(locale) = sys_locale::get_locale().as_ref() {
-        set_locale_inner(locale);
-    } else {
-        set_locale_inner(FALLBACK_LOCALE);
-    }
-}
-
-/// Get the current locale.
-pub fn locale() -> String {
-    rust_i18n::locale().to_string()
-}
+pub use i18n_embed::fluent::fluent_language_loader;
 
-/// Get all available locales.
-pub fn available_locales() -> Vec<&'static str> {
-    rust_i18n::available_locales!()
-}
+#[derive(RustEmbed)]
+#[folder = "i18n"]
+pub struct Localizations;
 
-fn set_locale_inner(locale: &str) {
-    let all_locales = rust_i18n::available_locales!();
-    if all_locales.contains(&locale) {
-        rust_i18n::set_locale(locale);
-    } else {
-        let language = split_locale(locale);
-        if all_locales.contains(&language.as_str()) {
-            rust_i18n::set_locale(&language);
-        } else {
-            rust_i18n::set_locale(FALLBACK_LOCALE);
-        }
-    }
-}
+pub static LANGUAGE_LOADER: OnceLock<FluentLanguageLoader> = OnceLock::new();
 
-fn split_locale(locale: &str) -> String {
-    let mut parts = locale.split(['-', '_']);
-    parts
-        .next()
-        .map_or_else(|| FALLBACK_LOCALE, |lang| lang)
-        .to_string()
-}
-
-// A macro for translating a text string.
 #[macro_export]
 macro_rules! t {
     ($($all:tt)*) => {
-        rust_i18n::t!($($all)*)
+        i18n_embed_fl::fl!($crate::locale::LANGUAGE_LOADER.get_or_init(|| $crate::locale::fluent_language_loader!()), $($all)*)
```

Comment:
@nc7s I modified this to work with `OnceLock`.

Reply:
This seems a bit all over the place since it's a macro. Maybe pack it into a function? (But then it's either a function call, or if a `const` function, still all over the place..)